src/OuterList.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace gapple\StructuredFields;
+
+class OuterList implements \IteratorAggregate, \ArrayAccess
+{
+    /**
+     * The array of values.
+     *
+     * @var array
+     */
+    public $value;
+
+    public function __construct($value = [])
+    {
+        array_walk($value, [$this, 'validateItemType']);
+
+        $this->value = $value;
+    }
+
+    private static function validateItemType($value): void
+    {
+        if (is_object($value)) {
+            if (!($value instanceof TupleInterface)) {
+                throw new \InvalidArgumentException();
+            }
+        } elseif (is_array($value)) {
+            if (count($value) != 2) {
+                throw new \InvalidArgumentException();
+            }
```

Comment:
Some additional validation could be done to ensure the tuple's parameters value is an object.
The value could be a scalar value, array, or object (`Bytes`, `Token`, or `Stringable`), so I don't think any validation is possible.

Reply:
I think best to leave detailed validation to the serializer where it's better defined.